WordPress/src/androidTest/java/org/wordpress/android/util/helpers/DynamicTextsProgressionHelperTest.kt

```diff
@@ -0,0 +1,250 @@
+package org.wordpress.android.util.helpers
+
+import android.view.ViewGroup.LayoutParams
+import android.widget.FrameLayout
+import android.widget.TextSwitcher
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isCompletelyDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.rule.ActivityTestRule
+import org.hamcrest.CoreMatchers.not
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.wordpress.android.R
+import org.wordpress.android.testing.TestActivity
+import java.lang.Thread.sleep
+import java.lang.ref.WeakReference
+
+class DynamicTextsProgressionHelperTest {
+    private val TEST_DISPLAY_DURATION: Long = 1000
+
+    @Rule @JvmField val rule = ActivityTestRule(TestActivity::class.java)
+
+    lateinit var textSwitcher: TextSwitcher
+    @Before
+    fun setUp() {
+        // create a text switcher as hte activity's view
+        rule.runOnUiThread {
+            TextSwitcher(rule.activity).also {
+                it.layoutParams = FrameLayout.LayoutParams(
+                        LayoutParams.MATCH_PARENT,
+                        LayoutParams.WRAP_CONTENT
+                )
+                rule.activity.setContentView(it)
+                it.id = R.id.progress_text
+                textSwitcher = it
+            }
+        }
+    }
+
+    @Test
+    fun that_helper_animates_texts() {
+        // setup
+        // (none)
+
+        rule.runOnUiThread {
+            DynamicTextsProgressionHelper(
+                    WeakReference(textSwitcher), listOf(
+                    R.string.notification_new_site_creation_creating_site_dynamic_subtitle_1,
+                    R.string.notification_new_site_creation_creating_site_dynamic_subtitle_2,
+                    R.string.notification_new_site_creation_creating_site_dynamic_subtitle_3,
+                    R.string.notification_new_site_creation_creating_site_dynamic_subtitle_4,
+                    R.string.notification_new_site_creation_creating_site_dynamic_subtitle_5,
+                    R.string.notification_new_site_creation_creating_site_dynamic_subtitle_6
+            ),
+                    TEST_DISPLAY_DURATION
+            ).also {
+                // execute
+
+                textSwitcher.postDelayed(it, it.delay)
+            }
+        }
+        // verify
+        onView(withText(R.string.notification_new_site_creation_creating_site_dynamic_subtitle_1))
```

Comment:
because there's only a `TextSwitcher` - every second the text on the screen should change - or the test will fail (`onView(withText(...))` will fail to match a view)